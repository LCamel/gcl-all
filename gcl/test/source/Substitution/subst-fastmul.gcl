con A, B : Int { A > 0 }
var x, y, z : Int
var even : Int -> Bool
var odd : Int -> Bool

{: INV = x * y + z = A * B :}

x, y, z := A, B, 0
{ INV ∧ x ≥ 0, bnd: x }
do x /= 0 ->
    z := z + y
    x := x - 1
od
{ z = A * B }
