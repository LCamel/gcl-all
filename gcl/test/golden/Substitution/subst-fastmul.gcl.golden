0: INV [ x,y,z \ A,B,0 ]
  INV [ x,y,z \ A,B,0 ] ===> A * B + 0 = A * B
    
1: INV [ x,y,z \ A,B,0 ]
  INV [ x,y,z \ A,B,0 ] ===> A * B + 0 = A * B
    
4: INV
  INV ===> x * y + z = A * B
    
11: INV [ x \ x - 1 ]
  INV [ x \ x - 1 ] ===> (x - 1) * y + z = A * B
    
12: INV [ x \ x - 1 ] [ z \ z + y ]
  INV [ x \ x - 1 ] [ z \ z + y ] ===> (x - 1) * y + z + y = A * B
    
15: INV
  INV ===> x * y + z = A * B
    
19: INV [ x \ x - 1 ] [ z \ z + y ]
  INV [ x \ x - 1 ] [ z \ z + y ] ===> (x - 1) * y + z + y = A * B
    
22: INV
  INV ===> x * y + z = A * B
    
30: INV
  INV ===> x * y + z = A * B
    